### ISSUE TITLE
Int and Decimal to Double conversion issues

### ISSUE THREAD

**Issue by SteveWilkes** [dadhi/FastExpressionCompiler#165]
Hello!

I'm seeing unexpected results with calls like these:

```C#
doubleProperty = (double)intValueHolder.Value;
doubleProperty = (double)decimalValueHolder.Value;
nullableDoubleProperty = (double?)decimalValueHolder.Value;
```

I'll add a PR with failing tests once I have this issue number.

Cheers,

Steve

**Comment by jogibear9988** [comment]
This one it's hard again, and will introduce more code for nullables...

Problem is conv decimal? => double
it's not included yet.

But I think we have to redo complete TryEmitConvert with nullable support in mind (For example implizit or explicit conversation with type or nullable type,...)

**Comment by dadhi** [comment]
Hi @SteveWilkes 
I have fixed these case, could you check v2.0.0-preview-02 ?

**Comment by SteveWilkes** [comment]
Will do, thanks very much! :)

**Comment by SteveWilkes** [comment]
These cases look good - thanks!

In numeric comparisons and conversions, I'm getting unexpected values with:

- Nullable int -> double
- Nullable decimal -> double
- Int -> enum

I'm getting 'Common Language Runtime detected an invalid program' with:

- Float -> decimal
- Float -> nullable decimal
- Double -> decimal

...and I'm getting null reference exceptions being thrown in `object.Equals()` calls comparing:

- Decimal with a constant decimal
- Nullable decimal with a constant decimal

I'll write failing tests for these cases - should I raise a new issue?

**Comment by dadhi** [comment]
No need, attach them here. The more tests, the more structured convert implementation will be. Thanks for writing.

**Comment by SteveWilkes** [comment]
Will do. Thanks! :)

**Comment by SteveWilkes** [comment]
Just an update - my unexpected value issues look to all have been due to a mistake I made moving the FEC source into my branch, so I think they're all solved!

**Comment by dadhi** [comment]
You mean everything is working except NRE?

**Comment by SteveWilkes** [comment]
No, I've still got the 'Common Language Runtime detected an invalid program' errors, one 'Attempted to read or write protected memory', and one issue with a comparison / assignment of a nullable long and nullable int. I'll at least add the last one of those under this issue.

**Comment by SteveWilkes** [comment]
Just to note - the 'Attempted to read or write protected memory' was another copy error on my part, so I've now raised all the issues I've mentioned here. Cheers!

**Comment by dadhi** [comment]
Thanks. 
Actually, I have narrowed down the problem to `TryEmitComparison` for nullables. Fix is in progress.
